### DISCUSSION TITLE
Unit 6.5 Video 4

### DISCUSSION THREAD

**Discussion by DSJohannes** [Lightning-AI/dl-fundamentals#48] [Lecture videos and quizzes] [answered]
Hello,

I've installed the "lightning training studio app" just like you described in the video. Every time I start the app I get this error message: 

mlp_cli2.py: error: Unrecognized arguments: model.learning_rate=0.010618472310873947 model.hidden_units=[50, 100] data.batch_size=32 trainer.max_epochs=1

followed by:

SystemExit: 2

So the app is terminating cause it can't recognize the passed arguments. I tried everything to solve this problem that came into my mind but that didn't help.
I'm running Python 3.10 in a Conda environment on MacOS Ventura with PyCharm.
Do you have any idea what the problem might be?

**Comment by rasbt** [comment, marked as answer]
Sorry for the hassle, my bad, I just saw the file that I uploaded had

```python
app = LightningApp(
    Sweep(
        script_path=ops.join(ops.dirname(__file__), "./mlp_cli2.py"),
        total_experiments=3,
        parallel_experiments=1,
        distributions={
            "model.learning_rate": LogUniform(0.001, 0.1),
            "model.hidden_units": Categorical(["[50, 100]", "[100, 200]"]),
            "data.batch_size": Categorical([32, 64]),
            "trainer.max_epochs": IntUniform(1, 3),
        },
        algorithm=OptunaAlgorithm(optuna.create_study(direction="maximize")),
        framework="pytorch_lightning",
    )
)
```

instead of 

```python
app = LightningApp(
    Sweep(
        script_path=ops.join(ops.dirname(__file__), "./mlp_cli2.py"),
        total_experiments=3,
        parallel_experiments=1,
        distributions={
            "--model.learning_rate": LogUniform(0.001, 0.1),
            "--model.hidden_units": Categorical(["[50, 100]", "[100, 200]"]),
            "--data.batch_size": Categorical([32, 64]),
            "--trainer.max_epochs": IntUniform(1, 3),
        },
        algorithm=OptunaAlgorithm(optuna.create_study(direction="maximize")),
        framework="pytorch_lightning",
    )
)
```

I.e., it was missing the `--`'s

**Reply by DSJohannes** [reply]
No problem, thank you very much, now it works fine.
